src/mpi/coll/reduce/reduce.c

```diff
@@ -216,7 +217,7 @@ int MPIR_Reduce_intra_auto (
     if ((count*type_size > MPIR_CVAR_REDUCE_SHORT_MSG_SIZE) &&
         (HANDLE_GET_KIND(op) == HANDLE_KIND_BUILTIN) && (count >= pof2)) {
         /* do a reduce-scatter followed by gather to root. */
-        mpi_errno = MPIR_Reduce_intra_reduce_scatter_gather(sendbuf, recvbuf, count, datatype, op, root, comm_ptr, errflag);
+        mpi_errno = MPIR_Reduce_intra_reduce_scatter_gather_const_space(sendbuf, recvbuf, count, datatype, op, root, comm_ptr, errflag);
```

Comment:
This essentially disables the previous implementation in favor of this new one. @pavanbalaji is that OK? Same goes for allreduce. Otherwise, I think this PR is ready to go.